source/particle/property/interface.cc

```diff
@@ -393,6 +393,40 @@ namespace aspect
         return (update & (update_default | update_values | update_gradients));
       }
 
+      template <int dim>
+      bool
+      Manager<dim>::plugin_name_exists(const std::string &name) const
+      {
+        return (std::find(plugin_names.begin(),plugin_names.end(),name) != plugin_names.end());
+      }
+
+      template <int dim>
+      bool
+      Manager<dim>::check_plugin_order(const std::string &first, const std::string &second) const
+      {
+        AssertThrow(plugin_name_exists(first),
+                    ExcMessage("Could not find a plugin with the name " + first + "."));
+        AssertThrow(plugin_name_exists(second),
+                    ExcMessage("Could not find a plugin with the name " + second + "."));
+
+        return (std::find(plugin_names.begin(),plugin_names.end(),first) < std::find(plugin_names.begin(),plugin_names.end(),second));
```

Comment:
```suggestion
        return (std::find(plugin_names.begin(),plugin_names.end(),first) 
                   < std::find(plugin_names.begin(),plugin_names.end(),second));
```

Reply:
What happens if `first == second`? Shouldn't we rule that out?

Reply:
hmm, `1 < 1 = false`, I agree an assert might be more appropriate.